src/main/java/gift/repository/MemoryProductRepository.java

```diff
@@ -0,0 +1,49 @@
+package gift.repository;
+
+import gift.dto.ProductResponseDto;
+import gift.entity.Product;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class MemoryProductRepository implements ProductRepository{
+    private final Map<Long, Product> productList = new HashMap<>();
+
+    public ProductResponseDto createProduct(Product product){
+        Long productId = productList.isEmpty() ? 1 : Collections.max(productList.keySet()) + 1;
+        product.setId(productId);
+
+        productList.put(productId, product);
+
+        return new ProductResponseDto(product);
+    }
+    public List<ProductResponseDto> searchAllProducts(){
```

Comment:
메서드와 메서드 사이에는 줄바꿈을 통해서 메서드의 가독성을 높여주시면 좋을 것 같아요. 😃 

```suggestion
    }
    
    public List<ProductResponseDto> searchAllProducts(){
```